contracts/Comptroller/Diamond/facets/SetterFacet.sol

```diff
@@ -0,0 +1,526 @@
+        ensureNonzeroAddress(newTreasuryAddress);
+
+        address oldTreasuryGuardian = treasuryGuardian;
+        address oldTreasuryAddress = treasuryAddress;
+        uint256 oldTreasuryPercent = treasuryPercent;
+
+        treasuryGuardian = newTreasuryGuardian;
+        treasuryAddress = newTreasuryAddress;
+        treasuryPercent = newTreasuryPercent;
+
+        emit NewTreasuryGuardian(oldTreasuryGuardian, newTreasuryGuardian);
+        emit NewTreasuryAddress(oldTreasuryAddress, newTreasuryAddress);
+        emit NewTreasuryPercent(oldTreasuryPercent, newTreasuryPercent);
+
+        return uint256(Error.NO_ERROR);
+    }
+
+    /*** Venus Distribution ***/
+
+    /**
+     * @dev Set ComptrollerLens contract address
+     * @param comptrollerLens_ The new ComptrollerLens contract address to be set
+     * @return uint256 0=success, otherwise a failure (see ErrorReporter.sol for details)
+     */
+    function _setComptrollerLens(
+        ComptrollerLensInterface comptrollerLens_
+    ) external compareAddress(address(comptrollerLens), address(comptrollerLens_)) returns (uint256) {
+        ensureAdmin();
+        ensureNonzeroAddress(address(comptrollerLens_));
+        address oldComptrollerLens = address(comptrollerLens);
+        comptrollerLens = comptrollerLens_;
+        emit NewComptrollerLens(oldComptrollerLens, address(comptrollerLens));
+
+        return uint256(Error.NO_ERROR);
+    }
+
+    /**
+     * @notice Set the amount of XVS distributed per block to VAI Vault
+     * @param venusVAIVaultRate_ The amount of XVS wei per block to distribute to VAI Vault
+     */
+    function _setVenusVAIVaultRate(
+        uint256 venusVAIVaultRate_
+    ) external compareValue(venusVAIVaultRate, venusVAIVaultRate_) {
+        ensureAdmin();
+        if (vaiVaultAddress != address(0)) {
+            releaseToVault();
+        }
+        uint256 oldVenusVAIVaultRate = venusVAIVaultRate;
+        venusVAIVaultRate = venusVAIVaultRate_;
+        emit NewVenusVAIVaultRate(oldVenusVAIVaultRate, venusVAIVaultRate_);
+    }
+
+    /**
+     * @notice Set the VAI Vault infos
+     * @param vault_ The address of the VAI Vault
+     * @param releaseStartBlock_ The start block of release to VAI Vault
+     * @param minReleaseAmount_ The minimum release amount to VAI Vault
+     */
+    function _setVAIVaultInfo(
+        address vault_,
+        uint256 releaseStartBlock_,
+        uint256 minReleaseAmount_
+    ) external compareAddress(vaiVaultAddress, vault_) {
+        ensureAdmin();
+        ensureNonzeroAddress(vault_);
+        if (vaiVaultAddress != address(0)) {
+            releaseToVault();
+        }
+
+        vaiVaultAddress = vault_;
+        releaseStartBlock = releaseStartBlock_;
+        minReleaseAmount = minReleaseAmount_;
+        emit NewVAIVaultInfo(vault_, releaseStartBlock_, minReleaseAmount_);
+    }
+
+    function _setForcedLiquidation(address vTokenBorrowed, bool enable) external {
```

Comment:
We need natspec docs here. And update this PR with the updated md file: https://github.com/VenusProtocol/venus-protocol-documentation/pull/69